### ISSUE TITLE
Expose flask/grpc tracer to enable child tracing in application logic

### ISSUE THREAD

**Issue by harshita-meena** [SeldonIO/seldon-core#4270]
Is there a better way or already available solution that can help user access the tracer object to add child traces for subcomponents of predictions. In the [hack I was trying in python library ](https://github.com/SeldonIO/seldon-core/compare/master...harshita-meena:seldon-core:patch-2?expand=1#diff-40636bc5e32d81e8351d317f73e59c5028dc5e4d0062c1e3e51ad5ec7c4e6ac2R101-R102), the user object can access it so it [can add child traces](https://github.com/harshita-meena/seldon-lgbm/compare/child-span?expand=1#diff-105974a25a6089510541e8ea2f47c72fd17b58d189d7db6c846dac2aa5079097R42-R44).  This helped enable component traces in the UI. One more question I have regarding jaeger traces is if it possible to set trace context by users ? I saw the traces are of type b3. 
<img width="781" alt="Screen Shot 2022-08-12 at 4 17 15 PM" src="https://user-images.githubusercontent.com/70981082/184455962-5c6ff9f8-2a2c-4bf0-a8a9-ff51afd81da8.png">

**Comment by cliveseldon** [comment]
Exposing the tracer to the user seems useful. Do you want to open a PR based on your work?

**Comment by harshita-meena** [comment]
Hi @cliveseldon i found a bug with grpc opentracing because of which i had to add a [hack](https://github.com/SeldonIO/seldon-core/pull/4300/files#diff-e0b992c1a0bb23ef86f5471a1a66a0a79afd26381ee4d6fe31e0d521f086ff34R240-R244), the issue is flasktracing differs from grpctracing in a way where the current context is not stored properly so instrumentation to access those traces does not work and any child trace created for a grpc tracer is not linked to its parent. 
I saw a similar [issue created by someone else ](https://github.com/opentracing-contrib/python-grpc/issues/34)on that repo but developers seem to be inactive. 

In short the current context is not getting set and I am trying to set it explicitly in predict function so it works well in [child trace creation](https://github.com/harshita-meena/seldon-lgbm/compare/child-span?expand=1#diff-105974a25a6089510541e8ea2f47c72fd17b58d189d7db6c846dac2aa5079097R42-R44).(get_current_span was  not able to fetch the context because none is set in grpc opentracing library, it returns None)

Let me know how I can solve this problem in a better way, happy to chat more to discuss the issue.

**Comment by harshita-meena** [comment]
Hi @cliveseldon happy to discuss more if the explanation was confusing. Please let me know.

**Comment by axsaucedo** [comment]
As discussed in the PR the suggestion is to continue adding this functionality in the v2 python wrapper instead of the v1 python wrapper - this functionality will be discussed through this issue https://github.com/SeldonIO/MLServer/issues/39
